data/other/tsv_maker.py

```diff
@@ -0,0 +1,75 @@
+import os
+import pandas as pd
+
+this_file = os.path.basename(__file__)
+dir_of_this_file = os.path.basename(os.path.dirname(__file__))
+parent_dir_of_this_file = os.path.basename(os.path.dirname(os.path.dirname(__file__)))
+created_by = os.path.join(parent_dir_of_this_file, dir_of_this_file, this_file)
+this_file = os.path.basename(__file__)
+source = ' using the 2015 U.S. Lighting Market Characterization report from U.S. DOE office of EERE prepared by Navigant Consulting.'
+
+count_col_label = 'source_count'
+weight_col_label = 'source_weight'
+
+class Lighting():
+    
+    def __init__(self,projects):
+       self.projects = projects
+
+    def create_lighting_tsv(self):
+    	""" 
+    	Create a lighting housing characteristic.
+    	"""
+    	# Initialize dataframe
+    	cols = ['Option=100% Incandescent','Option=100% CFL','Option=100% LED','source_count','source_weight']
+    	self.df = pd.DataFrame(columns=cols)
+
+    	# Using Data from Table 4.2 on page 49 of the report. 
+    	# ASSUMPTION: Incandescent contains Incandescent, Halogen, High Intensity Discharge, Other
+    	# ASSUMPTION: CFL contains CFL and LFL
+    	# ASSUMPTION: LED contains LED
+    	self.df.loc[0,cols] = [0.516,0.414,0.07,0,0]
+
+    def enforce_float_format(self,df):
+        """Ensure that all non Dependency= Columuns are floats.
+        Args:
+            df (pandas.DataFrame): the dataframe that the non "Dependency=" columns are converted to floats
+        Returns:
+            df (pandas.DataFrame): the same dataframe as the input argument except with non dependency columns as floats
+        """
+        matching = [s for s in df.columns.values if not "Dependency=" in s]
+        for col in matching:
+            df[col] = df[col].astype(float)
+        return df
+
+    def export_and_tag(self, df, filepath, project):
```

Comment:
I feel like we should try to inherit the existing `TSVMaker` base class as much as possible. See [this](https://github.com/NREL/OpenStudio-BuildStock/blob/master/data/recs/2009/tsv_maker.py#L21) as an example.

Reply:
@joseph-robertson Thanks for the comments. I refactored the RECS 2009, RECS 2015, and the OtherSources tsv_makers. I added the capability to pull the data from s3. Let me know if this cleans things up.